cdap-docs/developer-manual/source/metadata/programmatic-metadata.rst

```diff
@@ -0,0 +1,85 @@
+.. meta::
+    :author: Cask Data, Inc.
+    :copyright: Copyright © 2018 Cask Data, Inc.
+
+.. _programmatic-metadata:
+
+==================================
+Metadata from Programs and Plugins
+==================================
+
+.. highlight:: java
+
+Metadata can be accessed from *Programs* and *Plugins*. Currently, reading metadata is not supported while CDAP is running in *cloud mode* but metadata can be added, updated or deleted in all modes.
```

Comment:
Can you add a newline after 120 characters? Makes it easier to review. 

Also, I don't think CDAP has a cloud mode. Perhaps change it to ...reading metadata from a program or a plugin is not supported when the program is running in the cloud....

Although, is that a true statement? Is it supported when you run the program on a remote Hadoop cluster, for example?

Reply:
You can read metadata while running on hadoop cluster (in-prem). We use service discovery to discover metadata service and retrieve metadata through that. 

By remote hadoop cluster you mean the newly added hadoop provisioner?

Reply:
Done the first 2 changes.